### PR TITLE
Enable console audit logs for docker

### LINKED ISSUE #42666: Make timestamp field name in audit log consistent
I ran Elasticsearch 7.1.0 in a Docker container with audit logging turned on. Consider the following sample log entry I generated:

```
{"type": "server", "timestamp": "2019-05-28T19:17:36,708+0000", "level": "INFO", "component": "o.e.x.s.a.l.LoggingAuditTrail", "cluster.name": "docker-cluster", "node.name": "68394c3e3bbe", "cluster.uuid": "BI0VZnPyRJuQWQzhuwO8sg", "node.id": "mQjSOd7bREOwl_I4IUaEtw",  "message": "event.action=\"anonymous_access_denied\" event.type=\"rest\" node.id=\"mQjSOd7bREOwl_I4IUaEtw\" origin.address=\"172.17.0.1:36744\" origin.type=\"rest\" request.id=\"2nj_ww26T0eODoilHXDPSA\" request.method=\"GET\" url.path=\"/\""  }
```

Notice that the name of the timestamp field is `timestamp`.

Now, I ran Elasticsearch 7.1.0 again, but by downloading and extracting it from a `.tar.gz` file instead of in a Docker container. Here's a sample audit log entry I generated:

```
{"@timestamp":"2019-05-28T12:33:31,246", "node.id":"qUZG15diSquz8jpvQbTmhA", "event.type":"rest", "event.action":"anonymous_access_denied", "origin.type":"rest", "origin.address":"[::1]:49499", "url.path":"/", "request.method":"GET", "request.id":"b5A_F-ihQaOHrxBgAa3gjA"}
```

Notice that the name of the timestamp field is `@timestamp`.

This field name discrepancy is causing [some issues for users](https://discuss.elastic.co/t/filebeat-7-1-0-elasticsearch-module-audit-pipeline-incorrectly-references-timestamp-instead-of-timestamp/182721) trying to use Filebeat to parse the audit log, depending on whether the environment is a Docker container vs. not.

Other types of Elasticsearch logs (server, deprecation) consistently use `timestamp` (no `@` prefix) in either Docker or non-Docker logs.

For the short term we should make Filebeat look for either `@timestamp` or `timestamp` when its trying to parse Elasticsearch deprecation logs. I've created https://github.com/elastic/beats/issues/12339 to track this.

But for the longer term it would be nice if Elasticsearch consistently just used `timestamp` (no `@` prefix) for audit logs in both Docker and non-Docker logs.

### PR DESCRIPTION
**Pull request by pgomulka**
Enable audit logs in docker by creating console appenders for audit loggers.
also rename field @timestamp to `timestamp` and add field `type` with value `audit`

closes #42666

### CODE REVIEW

**Comment by pgomulka** [comment]
I decided not to include cluster.uuid in audit logs yet. I think we can try to refactor to use the same mechanism for populating cluster.uuid and node.id later. Possibly refactor to use the same ESJsonLayout (after I add a way to extend it - this is already WIP #41354 )

**Comment by pgomulka** [comment]
@elasticmachine ok to test

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java [hunk 1]
```diff
@@ -1075,7 +1075,9 @@ private void assertMsg(Logger logger, Map<String, String> checkFields, Map<Strin
                 logLine = logEntryFieldPattern.matcher(logLine).replaceFirst("");
             }
         }
-        logLine = logLine.replaceFirst("\"@timestamp\":\"[^\"]*\"", "").replaceAll("[{},]", "");
+        logLine = logLine.replaceFirst("\""+LoggingAuditTrail.LOG_TYPE+"\":\"audit\", ", "")
+                         .replaceFirst("\""+LoggingAuditTrail.TIMESTAMP+"\":\"[^\"]*\"", "")
```
Comment (jasontedor) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrailTests.java]: Sorry to nit, can you add spaces around the `+`s?

File: distribution/docker/build.gradle [hunk 2]
```diff
@@ -45,6 +45,7 @@ project.ext {
 
       into('config') {
         from project.projectDir.toPath().resolve("src/docker/config")
+        from project.projectDir.toPath().resolve(oss ? "oss-docker-build-context" : "docker-build-context").resolve("src/docker/config")
       }
```
Comment (jasontedor) [inline distribution/docker/build.gradle]: This is sneaky, can you add a comment explaining?

**Review by jasontedor — COMMENTED** [review]
I left some minor comments.

File: x-pack/plugin/core/src/main/config/log4j2.properties [hunk 3]
```diff
@@ -3,7 +3,8 @@ appender.audit_rolling.name = audit_rolling
 appender.audit_rolling.fileName = ${sys:es.logs.base_path}${sys:file.separator}${sys:es.logs.cluster_name}_audit.json
 appender.audit_rolling.layout.type = PatternLayout
 appender.audit_rolling.layout.pattern = {\
-                "@timestamp":"%d{ISO8601}"\
+                "type":"audit", \
+                "timestamp":"%d{yyyy-MM-dd'T'HH:mm:ss,SSSZ}"\
```
Comment (pgomulka) [inline x-pack/plugin/core/src/main/config/log4j2.properties]: @albertzaharovits in ES server we decided to add timezone to timestamps in logs. I included this change to audit logs too, so these are consistent
discussion link:
https://github.com/elastic/elasticsearch/pull/36833#discussion_r244225243

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by pgomulka** [comment]
@elasticmachine ok to test

**Pull request merged**